### PR TITLE
cli: print usage message on flag parsing errors

### LINKED ISSUE #6776: cli: cockroach quit doesn't work if you include unnecessary flags
Please follow the steps below to help us help you.
1. Please supply the header (i.e. the first few lines) of your most recent
   log file **for each node in your cluster**. On most unix-based systems
   running with defaults, this boils down to the output of

```
JESSEs-MacBook-Pro:cockroach jesseseldess$ grep -F '[config]' cockroach-data/logs/cockroach.INFO
I160518 11:55:50.508106 util/log/clog.go:998  [config] file created at: 2016/05/18 11:55:50
I160518 11:55:50.508106 util/log/clog.go:998  [config] running on machine: JESSEs-MacBook-Pro
I160518 11:55:50.508106 util/log/clog.go:998  [config] binary: CockroachDB beta-20160512-121-g7332e99 (darwin amd64, built 2016/05/16 20:22:13, go1.6)
I160518 11:55:50.508106 util/log/clog.go:998  [config] arguments: [./cockroach start --background --background=false]
```
1. Please describe the issue you observed:
- What did you do?

I started up a local node. Then I tried to run `quit` with an unnecessary flag (only `--http-port` is required).
- What did you expect to see?

When including the required flag (`--http-port`) and an unnecessary flag, I expected `quit` to work. When including only an unnecessary flag, I expected a helpful error instructing me to use the `--http-port` flag.
- What did you see instead?

In both of the above cases, I received an `unknown flag` error and `Failed running "quit"`:

```
JESSEs-MacBook-Pro:cockroach jesseseldess$ ./cockroach start
build:     beta-20160512-121-g7332e99 @ 2016/05/16 20:22:13 (go1.6)
admin:     http://localhost:8080
sql:       postgresql://root@localhost:26257?sslmode=disable
logs:      cockroach-data/logs
store[0]:  path=cockroach-data

JESSEs-MacBook-Pro:cockroach jesseseldess$ ./cockroach quit --port=26257
Error: unknown flag: --port
Failed running "quit"

JESSEs-MacBook-Pro:cockroach jesseseldess$ ./cockroach quit --store=cockroach-data
Error: unknown flag: --store
Failed running "quit"

JESSEs-MacBook-Pro:cockroach jesseseldess$ ./cockroach quit --http-port=8080 --port=26257
Error: unknown flag: --port
Failed running "quit"
```

### PR DESCRIPTION
**Pull request by spencerkimball**
Fixes #6776

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/14519)

### CODE REVIEW

**Review by petermattis — APPROVED** [review]
LGTM

**Comment by benesch** [comment]
CI failure looked transient; triggered another build.

File: pkg/cli/cli.go [hunk 1]
```diff
@@ -96,6 +96,15 @@ var isInteractive = isatty.IsTerminal(os.Stdout.Fd()) &&
 func init() {
 	cobra.EnableCommandSorting = false
 
+	// Set an error function for flag parsing which prints the usage message.
+	cockroachCmd.SetFlagErrorFunc(func(c *cobra.Command, err error) error {
+		if usageErr := c.Usage(); usageErr != nil {
```
Comment (tamird) [inline pkg/cli/cli.go]: there is no reason to call this `usageErr`; it should just be `err`

File: pkg/cli/cli.go [hunk 2]
```diff
@@ -96,6 +96,15 @@ var isInteractive = isatty.IsTerminal(os.Stdout.Fd()) &&
...
+			return usageErr
+		}
+		fmt.Println()
```
Comment (tamird) [inline pkg/cli/cli.go]: what is this?

**Review by tamird — COMMENTED** [review]
This needs a basic CLI example or test.

File: pkg/cli/cli.go (hunk 1 above)
Comment (spencerkimball) [inline pkg/cli/cli.go]: Done.

File: pkg/cli/cli.go (hunk 2 above)
Comment (spencerkimball) [inline pkg/cli/cli.go]: It's otherwise very difficult to see the actual error after the usage is printed because the usage is so busy. Added a comment.

**Pull request merged**